SingularityUI/app/components/tasks/Columns.jsx

```diff
@@ -0,0 +1,302 @@
+        Utils.timeStampFromNow(cellData)
+      )
+    }
+    sortable={true}
+  />
+);
+
+export const Host = (
+  <Column
+    label="Host"
+    id="host"
+    key="host"
+    cellData={
+      (rowData) => rowData.taskId.host
+    }
+    cellRender={
+      (cellData) => (
+        <a href={`${config.appRoot}/tasks/active/all/${cellData}`}>
+          {cellData}
+        </a>
+      )
+    }
+    sortable={true}
+  />
+);
+
+export const Rack = (
+  <Column
+    label="Rack"
+    id="rack"
+    key="rack"
+    cellData={
+      (rowData) => rowData.taskId.rackId
+    }
+    cellRender={
+      (cellData) => (
+        <a href={`${config.appRoot}/tasks/active/all/${cellData}`}>
+          {cellData}
+        </a>
+      )
+    }
+    sortable={true}
+  />
+);
+
+export const CPUs = (
+  <Column
+    label="CPUs"
+    id="cpus"
+    key="cpus"
+    cellData={
+      (rowData) => _.find(rowData.mesosTask.resources, (r) => r.name == 'cpus').scalar.value
+    }
+    cellRender={
+      (cellData) => (
+        <span>{cellData}</span>
+      )
+    }
+    sortable={true}
+  />
+);
+
+export const Memory = (
+  <Column
+    label="Memory"
+    id="memory"
+    key="memory"
+    cellData={
+      (rowData) => _.find(rowData.mesosTask.resources, (r) => r.name == 'mem').scalar.value
+    }
+    cellRender={
+      (cellData) => (
+        <span>{cellData} MB</span>
+      )
+    }
+    sortable={true}
+  />
+);
+
+export const ActiveActions = (onTaskKill) => {
+  return <Column
+    label=""
+    id="actions"
+    key="actions"
+    className="actions-column"
+    cellData={
+      (rowData) => rowData
+    }
+    cellRender={(cellData) => {
+        return (
+          <div className="hidden-xs">
+            <a data-action="remove" onClick={() => onTaskKill(cellData.taskId.id)}><Glyphicon iconClass="remove" /></a>
+            <JSONButton className="inline" object={cellData}>
+              {'{ }'}
+            </JSONButton>
+          </div>
+        );
+      }
+    }
+  />
+};
+
+export const NextRun = (
+  <Column
+    label="Next Run"
+    id="nextRun"
+    key="nextRun"
+    cellData={
+      (rowData) => rowData.pendingTask.pendingTaskId.nextRunAt
+    }
+    cellRender={(cellData) => {
+        let label = <span className="label label-default">SCHEDULED</span>;
+        if (Utils.timestampWithinSecionds(cellData, config.pendingWithinSeconds)) {
+          label = <span className="label label-info">PENDING</span>
+        } else if (cellData < Date.now() - config.pendingWithinSeconds * 1000) {
+          label = <span className="label label-danger">OVERDUE</span>
+        }
+        return (
+          <div>
+            {Utils.timeStampFromNow(cellData)} {label}
+          </div>
+        );
+      }
+    }
+    sortable={true}
+  />
+);
+
+export const PendingType = (
+  <Column
+    label="Pending Type"
+    id="pendingType"
+    key="pendingType"
+    cellData={
+      (rowData) => rowData.pendingTask.pendingTaskId.pendingType
```

Comment:
As far as I know that data will always exist for a pending task returned by the API. Maybe someone can correct me if I'm wrong.